src/rsvc/srv.c

```diff
@@ -1153,7 +1153,7 @@ bcast_create(crt_opcode_t opc, bool filter_invert, d_rank_list_t *filter_ranks,
 				    NULL /* co_bulk_hdl */, NULL /* priv */,
 				    filter_invert ?
 				    CRT_RPC_FLAG_FILTER_INVERT : 0,
-				    crt_tree_topo(CRT_TREE_FLAT, 0), rpc);
+				    crt_tree_topo(CRT_TREE_KNOMIAL, 0), rpc);
```

Comment:
[Request changes] According to the comment for `crt_tree_topo`, `branch_ratio` can't be `0`. @frostedcmos, could you suggest how we should choose a value for `branch_ratio`, please?

Reply:
According to the source comments:
"1075  * \param[in] branch_ratio     branch ratio, be ignored for CRT_TREE_FLAT.
1076  *                             for KNOMIAL tree or KARY tree, the valid value
1077  *                             should within the range of
1078  *                             [CRT_TREE_MIN_RATIO, CRT_TREE_MAX_RATIO], or
1079  *                             will be treated as invalid parameter.
"
Which puts branch factor valid values between 2 and 64.

This check however is not enforced in the crt_tree_topo(enum crt_tree_type tree_type, uint32_t branch_ratio) and only mentioned in the comments.

While this gives us a valid range [2,64], picking the optimal value from it is more challenging.

@liuxuezhao Do you have any suggestion on the question from @liw above?

Reply:
The current usage of ‘branch_ratio’ parameter passing：
iv_init:                             crt_tree_topo(CRT_TREE_KNOMIAL, 4)
create & destory rank:   crt_tree_topo(CRT_TREE_KNOMIAL, 4)
map_update:                  crt_tree_topo(CRT_TREE_KNOMIAL, 32)
pool_discard:                  crt_tree_topo(CRT_TREE_KNOMIAL, 32)

Reply:
Is it okay to use this parameter ’branch_ratio‘ as the specified configuration item in the configuration file? The default value is 4?

Reply:
sorry did not notice the question before.
crt_tree_topo(CRT_TREE_KNOMIAL, 0) is invalid, and in code crt_corpc_req_create() -> crt_tree_topo_valid(tree_topo) the check should return -DER_INVAL, no?
I tested the  "crt_tree_topo_valid(crt_tree_topo(CRT_TREE_KNOMIAL, 0))" can report invalid, the CI test did not report failure?

Reply:
> [Request changes] According to the comment for `crt_tree_topo`, `branch_ratio` can't be `0`. @frostedcmos, could you suggest how we should choose a value for `branch_ratio`, please?

here the bcast is for replica servers, the number of replica servers should be no too many? commonly is 3 ~ 5 or slightly more? so the original flat tree probably is fine, or use crt_tree_topo(CRT_TREE_KNOMIAL, 2).

I am not sure if should config it in yml file, as different place of code may bcast to different number of target servers.

Reply:
Which one is more recommended ？ keep it the same as before or (CRT_TREE_KNOMIAL, 2) ？

Reply:
(CRT_TREE_KNOMIAL, 2) probably is better than flat

Reply:
@liw @liuxuezhao hello, How can this PR land? As far as I know,  non intel contributors can't run all the CI，can this setting be adjusted？

Reply:
Hi, @yyyshi. AFAIK, this security policy can't be adjusted at the moment; with the establishment of the DAOS Foundation, there might be some new process that will make it easier for everyone (I hope). The CI job for this PR failed due to some unrelated issue. Could you merge with the latest daos-stack/daos master, so that we can get another CI run, please?

Reply:
> Hi, @yyyshi. AFAIK, this security policy can't be adjusted at the moment; with the establishment of the DAOS Foundation, there might be some new process that will make it easier for everyone (I hope). The CI job for this PR failed due to some unrelated issue. Could you merge with the latest daos-stack/daos master, so that we can get another CI run, please?

Of course i can, but it seems still failed in 'clang-format / Clang Format (pull_request) Failing after 1m' stage ?

Reply:
The clang-format check is only advisory. But the Jenkins-based tests are not running---I need to ask around.

Reply:
I have been advised to wait for https://github.com/daos-stack/daos/pull/13341 to land. So it sounds like there's a CI glitch.